File: step-function-map-in-parallel/sample_data_sets/parallel_generate.js
#!/usr/bin/env node

'use strict';

const fs = require('fs');
const util = require('util');

class Request {
    constructor(caseId) {
        this.caseId = caseId;
    }
}

class RequestSlot {
    constructor(requests) {
        this.requests = requests;
    }
}

class RequestMessage {
    constructor(requestId, requestSlots) {
        this.requestId = requestId;
        this.requestSlots = requestSlots;
    }
}

function main() {
    var targets = [1, 5, 10, 20, 40, 100, 200, 500, 700, 1000];
    var slot = 3;

    targets.forEach(function (target) {
        var requestSlots = [];

        var counter = 0;

        for (var targetSlot = 0; targetSlot < slot; targetSlot++) {
            var requests = []
            var requestSlot = new RequestSlot(requests);

            // Inject current slot into slot array
            requestSlots.push(requestSlot);
        }

        // Loop through all numbers for current slot
        while (counter < target) {
            var targetSlot = parseInt(counter / slot) % slot;
            requestSlots[targetSlot].requests.push(new Request(util.format("%d", counter)));
            counter++
        }

        var requestMessage = new RequestMessage('123456', requestSlots);

        var content = JSON.stringify(requestMessage);

        console.log("");
        console.log(util.format("Sample Dataset for %d cases:", target));
        console.log(content);

        try {
            fs.writeFileSync(util.format('dataset_%d.json', target), content);
            //file written successfully
        } catch (err) {
            console.error(err)
        }
    });
}

main();

File: step-function-map-in-parallel/sample_data_sets/parallel_generate.go
package main

import (
        "fmt"
        "encoding/json"
        "os"
        "log"
)

type Request struct {
        CaseId string `json:"caseId"`
}

type RequestSlot struct {
        Requests []Request `json:"requests"`
}

type RequestMessage struct {
        RequestId string `json:"requestId"`
        RequestSlots []RequestSlot `json:"requestSlots"`
}

func main() {
        targets := []int{1, 5, 10, 20, 40, 100, 200, 500, 700, 1000}
        slot := 3
        
        for _,target := range targets {

                requestSlots := make([]RequestSlot, 0)

                for xi := 0; xi < slot; xi++ {
                        requests := make([]Request, 0)
                        requestSlot := RequestSlot{requests}
                        // Inject current slot into slot array
                        requestSlots = append(requestSlots, requestSlot)
                }

                // Loop through all numbers in target
                for counter := 0; counter < target; counter++ {
                        // Divide by slot and get reminder of result (to ensure not exceed slot count) to get into correct slot
                        targetSlotIndex := (counter / slot) % slot
                        requestSlots[targetSlotIndex].Requests = append(requestSlots[targetSlotIndex].Requests, Request{fmt.Sprintf("%d", counter)})
                }

                m := RequestMessage{"123456", requestSlots}

                b, _ := json.Marshal(m)

                fmt.Println("")
                fmt.Println(fmt.Sprintf("Sample Dataset for %d cases:", target))
                fmt.Println(string(b))

                err := os.WriteFile(fmt.Sprintf("dataset_%d.json", target), []byte(string(b)), 0644)
                if err != nil {
                                log.Fatal(err)
                }
        }
}